packages/contentstack-export-to-csv/src/util/index.js

```diff
@@ -678,6 +683,174 @@ function wait(time) {
+    .roles()
+    .then((roles) => {
+      roles.items.forEach((item) => {
+        if (item.name === 'member' || item.name === 'admin') {
+          roleMap.name = item.uid;
+        }
+      });
+    })
+    .catch((err) => {
+      handleErrorMsg(err);
+    });
+  return roleMap;
+}
+
+async function cleanTeamsData(data, managementAPIClient, org) {
+  const roleMap = await getOrgRoles(managementAPIClient, org);
+  const fieldToBeDeleted = [
+    '_id',
+    'createdAt',
+    'createdBy',
+    'updatedAt',
+    'updatedBy',
+    '__v',
+    'createdByUserName',
+    'updatedByUserName',
+    'organizationUid',
+  ];
+  if (data?.length !== 0) {
+    data.forEach((team) => {
+      fieldToBeDeleted.forEach((fields) => {
+        delete team[fields];
+      });
+      if (!team.hasOwnProperty('description')) {
+        team.description = '';
+      }
+      if (team.organizationRole === roleMap['member']) {
+        team.organizationRole = 'member';
+      } else {
+        team.organizationRole = 'admin';
+      }
+      team.Total_Members = team?.users?.length || 0;
+    });
+  } else {
+    cliux.print(`info: There are no teams in the given org`);
+  }
+  return data;
+}
+
+async function exportTeams(managementAPIClient, organization, teamUid) {
+  const allTeamsData = await exportOrgTeams(managementAPIClient, organization);
+  if (allTeamsData.length === 0) {
+    this.log(`There are not teams in the organization named ${organization?.name}`);
+  } else {
+    const modifiedTeam = _.cloneDeep(allTeamsData);
+    modifiedTeam.forEach((team) => {
+      delete team['users'];
+      delete team['stackRoleMapping'];
+    });
+    const fileName = `${kebabize(organization.name.replace(config.organizationNameRegex, ''))}_teams_export.csv`;
+    write(this, modifiedTeam, fileName, 'organization Team details');
+    // exporting teams user data or a single team user data
+    await getTeamsDetail(allTeamsData,organization,teamUid);
+  }
+}
+
+async function getTeamsDetail(allTeamsData, organization ,teamUid) {
+  if (!teamUid) {
+
+    const userData = await getTeamsUserDetails(allTeamsData);
+    const fileName = `${kebabize(
+      organization.name.replace(config.organizationNameRegex, ''),
+    )}_team_User_Details_export.csv`;
+
+    write(this, userData, fileName, 'Team User details');
+  
+  } else {
+
+    const team = allTeamsData.filter((team) => team.uid === teamUid)[0];
+    
+    team.users.forEach((user) => {
+      user['team-name'] = team.name;
+      user['team-uid'] = team.uid;
+      delete user['active'];
+      delete user['orgInvitationStatus'];
+    });
+
+    const fileName = `${kebabize(
+      organization.name.replace(config.organizationNameRegex, ''),
+    )}_team_${teamUid}_User_Details_export.csv`;
+    
+    write(this, team.users, fileName, 'Team User details');
+  
+  }
+}
+
+async function getTeamsUserDetails(teamsObject) {
+  const allTeamUsers = [];
+  teamsObject.forEach((team) => {
+    if (team.users.length) {
```

Comment:
@cs-raj, here we can use optional chaining.
```suggestion
    if (team.users.length) {
```

Reply:
Yes added